lib/filesystem/src/path.cpp

```diff
@@ -368,7 +368,7 @@ namespace irods::experimental::filesystem
             return *this;
         }
 
-        if (is_separator(fp[pos_ - 1])) {
+        if (pos_ != 0 && is_separator(fp[pos_ - 1])) {
```

Comment:
`> 0` instead of `!= 0`?

Reply:
Hmm, I guess `!=` is fine too since the impl is supposed to guard against range errors.

Reply:
Definitely need to run the tests to make sure this change doesn't break anything.

Reply:
Good to know. Can't remember if the check for 0 was intentionally left out. Either it's an oversight or it was determined unnecessary by design.

Reply:
Made this change following the UBSAN report if it helps:

```
...
/irods_source/lib/filesystem/src/path.cpp:371:34: runtime error: unsigned integer overflow: 0 - 1 cannot be represented in type 'unsigned long'
...
```

Reply:
Updated to `> 0`.